packages/app/src/firestore.rules.spec.js

```diff
@@ -0,0 +1,127 @@
+const firebase = require('@firebase/testing')
+const path = require('path')
+const fs = require('fs')
+const util = require('util')
+const readFileAsync = util.promisify(fs.readFile)
+
+const projectId = 'pair-roulette'
+
+describe('Firestore rules', () => {
+  beforeAll(async () => {
+    const rulesPath = path.join(__dirname, 'firestore.rules')
+    const rules = await readFileAsync(rulesPath, 'utf8')
+    await firebase.loadFirestoreRules({ projectId, rules })
+  })
+
+  beforeEach(async () => {
+    await firebase.clearFirestoreData({ projectId })
+  })
+
+  afterAll(async () => {
+    await Promise.all(firebase.apps().map(app => app.delete()))
+
+    const coverageUrl = `http://localhost:8080/emulator/v1/projects/${projectId}:ruleCoverage.html`
+    console.log(`View rule coverage information at ${coverageUrl}\n`)
+  })
+
+  it('require users to log in before creating a profile', async () => {
+    const anonymous = authedApp(null)
+
+    const profile = createProfile(anonymous, 'alice')
+
+    await firebase.assertFails(profile)
+  })
+
+  it('should enforce the createdAt date in user profiles', async () => {
+    const uid = 'alice'
+    const alice = authedApp({ uid })
+
+    const profile = alice.collection('users').doc(uid).set({ uid })
+
+    await firebase.assertFails(profile)
+  })
+
+  it('should only let users create their own profile', async () => {
+    const alice = authedApp({ uid: 'alice' })
+
+    const bobProfile = createProfile(alice, 'bob')
+
+    await firebase.assertFails(bobProfile)
+  })
+
+  it('should allow owner to read document', async () => {
+    const alice = authedApp({ uid: 'alice' })
+
+    await createProfile(alice, 'alice')
+
+    await firebase.assertSucceeds(getProfile(alice, 'alice'))
+  })
+
+  it('should only allow owner to read document', async () => {
+    const alice = authedApp({ uid: 'alice' })
+    const bob = authedApp({ uid: 'bob' })
+
+    await createProfile(alice, 'alice')
+
+    await firebase.assertFails(getProfile(bob, 'alice'))
+  })
+
+  it('should allow owner to update document', async () => {
+    const alice = authedApp({ uid: 'alice' })
+    await createProfile(alice, 'alice')
+
+    const profile = updateProfile(alice, 'alice')
+
+    await firebase.assertSucceeds(profile)
+  })
+
+  it('should enforce the modifiedAt date in user profiles', async () => {
+    const uid = 'alice'
+    const alice = authedApp({ uid })
+    await createProfile(alice, uid)
+
+    const profile = alice.collection('users').doc(uid).update({ uid })
+
+    await firebase.assertFails(profile)
+  })
+
+  it('should only let users update their own profile', async () => {
```

Comment:
Par contre on test jamais une mise à jour avec succès contrairement à la création.

Reply:
Hm, si. C'est le test suivant : `should allow owner to update document`.

Reply:
Ah oui pardon j'étais dans la logique test en erreur puis test en succès et du coup j'ai pas fait attention qu'il était avant. Alors top 👍